Add ome-zarr diagram

will-moore opened the pull request:
Interactive / cheat-sheet zarr diagram.

Preview at http://will-moore.github.io/presentations/2020/Dundee/zarr_diagram/index.html

will-moore committed:
Add ome-zarr diagram

mtbc commented on 2020/OME_Meeting/zarr_diagram/index.html:
```diff
@@ -0,0 +1,262 @@
+            <div id="omero-ms-zarr" class="hoverExpand">
+                <code><pre>$ git clone -b v0.1.1 git://github.com/ome/omero-ms-zarr
+$ gradle build -x test
+$ cat <<EOF > cfg
```
I'm puzzled as to why the preview is showing me, `$ cat < cfg`.

joshmoore replied:
Probably needs to use `&lt;

joshmoore commented:
A few miscellaneous ideas from driving:
 - could the links in the mouseovers be clickable?
 - how could we work in the new https://workshop.org/image  URL?
 - might be worth a bit of text saying "mouseover shows..."

will-moore committed:
Fixes to zarr diagram, Support server URL

will-moore commented:
Fixed the ```<< EOF >```
I added the server URL separate from the s3 URL.
Now, if you CLICK into either input, or edit the input, then the Dask/napari/ome-zarr URLs update to *that* URL. It's not obvious behaviour, so I guess there's a question about whether this is a presentation or are users going to try working it out on their own.
Same question for mouseovers. I don't really want to add much to the diagram since any clutter detracts from the clarity.

 For the URLs, the 2 in the ```omero-ms-zarr``` are clickable, but I wasn't sure what other URLs are meant to be clickable - e.g. endpoints in aws?
e.g. URLs like 'https://s3.embassy.ebi.ac.uk/6001240.zarr' that are permission denied?

joshmoore commented on 2020/OME_Meeting/zarr_diagram/index.html:
```diff
@@ -0,0 +1,301 @@
+            <div class="inputs" style="top: 123px; left:602px">
+                <!-- <label>server-url: </label><br/> -->
+                <br/>
+                <input id="server" placeholder="server-url" value="https://workshop.org/image" style="width:250px" />
```
```suggestion
                <input id="server" placeholder="server-url" value="https://workshop.openmicroscopy.org/image" style="width:250px" />
```

joshmoore reviewed (commented):
Barring the one fix, looks fine. Yeah, the click ability is a bit unclear at the moment. Let's show it to people and see what they say.

will-moore committed:
Fix workshop.openmicroscopy url

will-moore committed:
Tweak zarr diagram png lines

will-moore committed:
Move http inputs over lines

will-moore commented:
Updated to make it clearer that the URL for napari/dask loading data from s3 is different from omero-ms-zarr. Lines pass through the URL input itself.

joshmoore commented:
- I like the new location of the URLs.
 - But I'm failing to get the URLs to update when I set a new image ID or URL.
 - Assuming the name of the folder can't be updated, perhaps better to drop the `1.` from the name.
 - I'm a bit unclear on the black coming down from "ome-zarr" before "dask". I assume this represents that at some point it becomes numpy? I'm just not sure _when_ that happens myself. Maybe not too important.

will-moore committed:
Fix keyup event for imageid

jburel commented:
naming: The folder should be renamed ``Dundee`` otherwise when this is deployed we will end with 2 directories under 2020. see https://downloads.openmicroscopy.org/presentations/2020/Dundee/

will-moore committed:
Rename to 2020/Dundee/

joshmoore commented:
Download section is missing a slash: `ome_zarr download https://s3.embassy.ebi.ac.uk/idr/zarr/v0.11.zarr`

will-moore committed:
Fix missing / in download url

will-moore committed:
Rename folder 1.zarr -> .zarr in image

will-moore commented:
OK, I think I have fixed everything now in http://will-moore.github.io/presentations/2020/Dundee/zarr_diagram/index.html
except the ```black coming down from "ome-zarr" before "dask"```.

Black doesn't have a particular meaning. Blue just means "ome zarr format" which is what the ome-zarr gives you. Didn't try to encode any more meaning than that.

joshmoore commented:
LGTM. Thanks again.
